bridge_adaptivity/module/models.py

```diff
@@ -116,49 +140,42 @@ def get_absolute_url(self):
 class Engine(models.Model):
     """Defines engine settings."""
 
-    DEFAULT_ENGINE_NAME = 'mock'
+    DEFAULT_ENGINE = 'engine_mock'
+    DRIVER = None
 
-    name = models.CharField(default=DEFAULT_ENGINE_NAME, max_length=255)
+    engine = models.CharField(choices=ENGINES, default=DEFAULT_ENGINE, max_length=100)
+    engine_name = models.CharField(max_length=255, blank=True, null=True, unique=True)
     host = models.URLField(blank=True, null=True)
     token = models.CharField(max_length=255, blank=True, null=True)
-
-    _engines_cache = {}
+    is_default = fields.BooleanField(default=False, help_text=_("If checked Engine will be used as the default!"))
 
     class Meta:
         unique_together = ('host', 'token')
 
     def __str__(self):
-        return "Engine: {}".format(self.name)
+        return "Engine: {}".format(self.engine_name)
+
+    def save(self, *args, **kwargs):
+        utils.save_model_parameter_true_once(self, 'is_default')
```

Comment:
little bit not clear function name as for me. 

I understand that you save model with flag is_default=True and then update all other objects in DB with is_default=False, but I understand it only after opening this function and reading code.

In my branch I created such mixing:
```
class ModelWithDefaultInstanceMixin(object):
    IS_DEFAULT_FIELD = 'is_default'

    def save(self, *args, **kwargs):
        if not hasattr(self, self.IS_DEFAULT_FIELD):
            raise ValueError('Model {} has no field {}'.format(self, self.IS_DEFAULT_FIELD))

        if getattr(self, self.IS_DEFAULT_FIELD, None):
            default_qs = self.__class__.objects.filter(**{self.IS_DEFAULT_FIELD: True})
            if default_qs:
                default_qs.update(is_default=False)
        return super(ModelWithDefaultInstanceMixin, self).save(*args, **kwargs)
```

don't you think it's better solution?

Reply:
And using this mixin you don't need to define save for each model as you do here in this PR.

Reply:
Mixin works for me. We will improve this code while merging with your branch.